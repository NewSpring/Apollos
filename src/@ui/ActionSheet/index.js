export { ActionSheetProvider, connectActionSheet } from '@expo/react-native-action-sheet';
