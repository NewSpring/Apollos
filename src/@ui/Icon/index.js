import PropTypes from 'prop-types';
import React from 'react';
import { compose, pure, setPropTypes } from 'recompose';
import { flow, camelCase, upperFirst, kebabCase } from 'lodash';

import Placeholder from '@ui/Placeholder';
import * as Icons from './icons';

const pascalCase = string => flow(camelCase, upperFirst)(string);

// Convenience component to render icons based on the icon's string name, like:
// <Icon name="skip-next" {...allOtherPropsPassedToComponent} />
//
// Can also import the icon directly:
// import { SkipNext } from 'Icon/icons';
// <SkipNext />

const enhance = compose(
  pure,
  setPropTypes({
    name: PropTypes.oneOf(Object.keys(Icons).map(kebabCase)).isRequired,
    size: PropTypes.number,
    fill: PropTypes.string,
    isLoading: PropTypes.bool,
  }),
);

const Icon = enhance(({
  name,
  size,
  isLoading = false,
  ...otherProps
}) => {
  const IconComponent = Icons[pascalCase(name)];
  return (
    <Placeholder.Media
      size={size === undefined ? 32 : size} // 32 is the default size used within the svg component
      hasRadius
      onReady={!isLoading}
    >
      <IconComponent size={size} {...otherProps} />
    </Placeholder.Media>
  );
});

export default Icon;
