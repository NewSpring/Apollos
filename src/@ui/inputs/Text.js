import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { TextInput, Animated, Easing } from 'react-native';
import styled from '@ui/styled';

import FloatingLabel from './FloatingLabel';
import InputUnderline from './InputUnderline';
import InputAddon from './InputAddon';
import InputWrapper, { InputRow } from './InputWrapper';

const StyledTextInput = styled(({ theme }) => ({
  paddingVertical: theme.sizing.baseUnit / 4,
}), 'Input.Text.TextInput')(TextInput);

class Text extends PureComponent {
  static propTypes = {
    label: PropTypes.string.isRequired,
    animationDuration: PropTypes.number,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    prefix: PropTypes.node,
    suffix: PropTypes.node,
  };

  static defaultProps = {
    animationDuration: 450,
  }

  focusAnimation = new Animated.Value(0);

  playAnimation = (toValue) => {
    Animated.timing(this.focusAnimation, {
      toValue,
      duration: this.props.animationDuration,
      easing: Easing.in(Easing.bezier(0.23, 1, 0.32, 1)),
      useNativeDriver: true,
    }).start();
  };

  handleFocus = (...args) => {
    if (this.props.onFocus) this.props.onFocus(...args);
    this.playAnimation(1);
  }

  handleBlur = (...args) => {
    if (this.props.onBlur) this.props.onBlur(...args);
    this.playAnimation(0);
  }

  render() {
    const {
      label,
      animationDuration,
      prefix,
      suffix,
      ...textInputProps
    } = this.props;
    return (
      <InputWrapper>
        <InputRow>
          <InputAddon>{this.props.prefix}</InputAddon>
          <Animated.View style={{ opacity: this.focusAnimation, flex: 1 }}>
            <StyledTextInput
              {...textInputProps}
              onFocus={this.handleFocus}
              onBlur={this.handleBlur}
            />
          </Animated.View>
          <InputAddon>{this.props.suffix}</InputAddon>
        </InputRow>
        <FloatingLabel animation={this.focusAnimation}>{label}</FloatingLabel>
        <InputUnderline animation={this.focusAnimation} />
      </InputWrapper>
    );
  }
}

export default Text;
