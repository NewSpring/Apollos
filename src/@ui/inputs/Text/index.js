import React from 'react';
import PropTypes from 'prop-types';
import { get } from 'lodash';
import { compose, withProps, pure } from 'recompose';
import { TextInput, Animated } from 'react-native';

import createInput from '../createInput';
import InputAddon, { AddonRow } from '../InputAddon';
import withInputControlStyles from '../withInputControlStyles';

const StyledTextInput = withInputControlStyles(TextInput);

const propsForInputType = {
  password: {
    secureTextEntry: true,
    autoCapitalize: 'none',
    autoCorrect: false,
  },
  email: {
    keyboardType: 'email-address',
    autoCapitalize: 'none',
    autoCorrect: false,
  },
  numeric: {
    keyboardType: 'numeric',
  },
  phone: {
    keyboardType: 'phone-pad',
  },
};

const enhance = compose(
  pure,
  createInput,
  withProps(({ type, ...props }) => ({
    ...get(propsForInputType, type, {}),
    ...props,
  })),
);

const Text = enhance(({
  prefix,
  suffix,
  focusAnimation, // from createInput
  ...textInputProps
}) => (
  <AddonRow>
    <InputAddon>{prefix}</InputAddon>
    <Animated.View style={{ opacity: focusAnimation, flex: 1 }}>
      <StyledTextInput {...textInputProps} />
    </Animated.View>
    <InputAddon>{suffix}</InputAddon>
  </AddonRow>
));

Text.propTypes = {
  prefix: PropTypes.node,
  suffix: PropTypes.node,
};

export default Text;
