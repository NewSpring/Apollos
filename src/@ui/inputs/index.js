export { default as Picker, Item as PickerItem } from './Picker';
export { default as Switch } from './Switch';
export { default as Text } from './Text';
export { default as DateInput } from './DateInput';
