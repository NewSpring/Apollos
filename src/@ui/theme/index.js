export { default as ThemeProvider } from './ThemeProvider';
export { default as withTheme } from './withTheme';
export { ThemeMixin, withThemeMixin } from './mixins';
