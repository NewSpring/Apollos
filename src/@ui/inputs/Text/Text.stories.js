import React from 'react';

import { storiesOf } from '@storybook/react-native';
import Icon from '@ui/Icon';
import FlexedView from '@ui/FlexedView';

import Text from './';

storiesOf('Inputs', module)
  .add('Text', () => (
    <FlexedView>
      <Text
        editable
        label="Some label text"
        placeholder="Some placeholder"
      />
      <Text
        editable
        type="password"
        label="Password"
        placeholder="Some placeholder"
        suffix={<Icon name="lock" size={18} />}
      />
      <Text
        editable
        type="email"
        label="Email"
        placeholder="Some placeholder"
      />
      <Text
        editable
        type="numeric"
        label="Numeric"
        placeholder="Some placeholder"
      />
      <Text
        editable
        type="phone"
        label="Phone Number"
        placeholder="Some placeholder"
      />
    </FlexedView>
  ));
