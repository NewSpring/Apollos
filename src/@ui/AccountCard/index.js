import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import { Platform } from 'react-native';
import { startCase, toLower } from 'lodash';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card, { CardContent } from '@ui/Card';
import SideBySideView from '@ui/SideBySideView';
import FlexedView from '@ui/FlexedView';
import { H5 } from '@ui/typography';
import CategoryLabel from '@ui/CategoryLabel';
import Icon from '@ui/Icon';
import last4 from '@utils/last4';
import { withTheme } from '@ui/theme';

const enhance = compose(
  setPropTypes({
    title: PropTypes.string.isRequired,
    isLoading: PropTypes.bool,
    accountNumber: PropTypes.number,
    accountType: PropTypes.number,
    iconSize: PropTypes.number,
  }),
  defaultProps({
    isLoading: false,
    accountNumber: '',
    accountType: 'creditCard',
    iconSize: undefined,
  }),
  withIsLoading,
  withTheme(({ theme, ...otherProps }) => ({
    iconSize: otherProps.iconSize || theme.helpers.rem(1),
  })),
  pure,
);

const HorizontalLayout = styled({
  alignItems: 'center',
  minHeight: 110, // kind of the best middle ground for various title lengths.
})(SideBySideView);

const LeftColumn = styled({
  flex: 1.66,
})(CardContent);

const RightColumn = styled(({ theme }) => ({
  paddingRight: theme.sizing.baseUnit,
  alignItems: 'flex-end',
  justifyContent: 'center',
  ...Platform.select({
    web: {
      position: 'relative',
    },
  }),
}))(FlexedView);

const AccountCard = enhance(({
  title,
  isLoading,
  accountNumber,
  accountType,
  iconSize,
  ...otherProps
}) => (
  <Card isLoading={isLoading} {...otherProps}>
    <HorizontalLayout>
      <LeftColumn>
        <H5>{startCase(toLower(title))}</H5>
        <CategoryLabel
          label={last4(accountNumber)}
          icon={accountType === 'bankAccount' ? 'bank' : 'credit'}
          isLoading={isLoading}
        />
      </LeftColumn>

      <RightColumn>
        <Icon name="ArrowNext" size={iconSize} />
      </RightColumn>
    </HorizontalLayout>
  </Card>
));

export default AccountCard;
