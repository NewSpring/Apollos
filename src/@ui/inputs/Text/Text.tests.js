import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@ui/theme';

import Icon from '@ui/Icon';
import Text from './';

describe('The Text Input component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Text
          editable
          label="Some label text"
          placeholder="Some placeholder"
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as a password field with an icon', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Text
          editable
          type="password"
          label="Password"
          placeholder="Some placeholder"
          suffix={<Icon name="lock" size={18} />}
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as an email field', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Text
          editable
          type="email"
          label="Email"
          placeholder="Some placeholder"
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as a numeric field', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Text
          editable
          type="numeric"
          label="Numeric"
          placeholder="Some placeholder"
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as a phone number field', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Text
          type="phone"
          label="Phone Number"
          placeholder="Some placeholder"
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
