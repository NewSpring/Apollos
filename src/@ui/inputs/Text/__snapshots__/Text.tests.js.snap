// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Text Input component should render as a numeric field 1`] = `
<View
  disabled={false}
  style={
    Object {
      "marginVertical": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "flex": 1,
            "opacity": 0,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          keyboardType="numeric"
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="Some placeholder"
          placeholderTextColor="rgba(48, 48, 48, 0.6)"
          returnKeyType="done"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 15.75,
              "height": 40,
              "lineHeight": 30,
              "paddingVertical": 5,
            }
          }
          type="numeric"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
    </View>
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#505050",
            "fontFamily": "Colfax-Bold",
            "fontSize": 15.75,
            "lineHeight": 17.75,
            "paddingBottom": 1.8,
            "paddingTop": 4.5,
            "paddingVertical": 5,
          }
        }
      >
        Numeric
      </Text>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#dddddd",
            "bottom": -0.5,
            "height": 0.5,
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
            "transform": Array [
              Object {
                "scaleX": 0.0001,
              },
            ],
          }
        }
      >
        <View
          hasError={false}
          style={
            Object {
              "backgroundColor": "#6bac43",
              "bottom": 0,
              "height": 0.5,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`The Text Input component should render as a password field with an icon 1`] = `
<View
  disabled={false}
  style={
    Object {
      "marginVertical": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "flex": 1,
            "opacity": 0,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          editable={true}
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="Some placeholder"
          placeholderTextColor="rgba(48, 48, 48, 0.6)"
          returnKeyType="done"
          secureTextEntry={true}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 15.75,
              "height": 40,
              "lineHeight": 30,
              "paddingVertical": 5,
            }
          }
          type="password"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Svg
          height={18}
          viewBox="0 0 24 24"
          width={18}
        >
          <Path
            d="M9 11.18h6.4V8.73c0-.9-.3-1.68-.94-2.32-.62-.6-1.38-.9-2.26-.9-.88 0-1.64.3-2.26.97C9.3 7.1 9 7.87 9 8.77v2.45zm10.4 1.23v7.4c0 .37-.12.66-.35.9-.23.24-.52.36-.85.36h-12c-.33 0-.62-.1-.85-.35-.23-.24-.35-.53-.35-.87v-7.4c0-.32.12-.6.35-.85.23-.24.52-.36.85-.36h.4V8.74c0-1.57.55-2.92 1.65-4.04C9.35 3.5 10.67 3 12.2 3c1.53 0 2.85.56 3.95 1.7 1.1 1.1 1.65 2.46 1.65 4.03v2.45h.4c.33 0 .62.12.85.36.23.24.35.53.35.87z"
            fill="#303030"
          />
        </Svg>
      </View>
    </View>
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#505050",
            "fontFamily": "Colfax-Bold",
            "fontSize": 15.75,
            "lineHeight": 17.75,
            "paddingBottom": 1.8,
            "paddingTop": 4.5,
            "paddingVertical": 5,
          }
        }
      >
        Password
      </Text>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#dddddd",
            "bottom": -0.5,
            "height": 0.5,
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
            "transform": Array [
              Object {
                "scaleX": 0.0001,
              },
            ],
          }
        }
      >
        <View
          hasError={false}
          style={
            Object {
              "backgroundColor": "#6bac43",
              "bottom": 0,
              "height": 0.5,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`The Text Input component should render as a phone number field 1`] = `
<View
  disabled={false}
  style={
    Object {
      "marginVertical": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "flex": 1,
            "opacity": 0,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          keyboardType="phone-pad"
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="Some placeholder"
          placeholderTextColor="rgba(48, 48, 48, 0.6)"
          returnKeyType="done"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 15.75,
              "height": 40,
              "lineHeight": 30,
              "paddingVertical": 5,
            }
          }
          type="phone"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
    </View>
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#505050",
            "fontFamily": "Colfax-Bold",
            "fontSize": 15.75,
            "lineHeight": 17.75,
            "paddingBottom": 1.8,
            "paddingTop": 4.5,
            "paddingVertical": 5,
          }
        }
      >
        Phone Number
      </Text>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#dddddd",
            "bottom": -0.5,
            "height": 0.5,
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
            "transform": Array [
              Object {
                "scaleX": 0.0001,
              },
            ],
          }
        }
      >
        <View
          hasError={false}
          style={
            Object {
              "backgroundColor": "#6bac43",
              "bottom": 0,
              "height": 0.5,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`The Text Input component should render as an email field 1`] = `
<View
  disabled={false}
  style={
    Object {
      "marginVertical": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "flex": 1,
            "opacity": 0,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          editable={true}
          keyboardType="email-address"
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="Some placeholder"
          placeholderTextColor="rgba(48, 48, 48, 0.6)"
          returnKeyType="done"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 15.75,
              "height": 40,
              "lineHeight": 30,
              "paddingVertical": 5,
            }
          }
          type="email"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
    </View>
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#505050",
            "fontFamily": "Colfax-Bold",
            "fontSize": 15.75,
            "lineHeight": 17.75,
            "paddingBottom": 1.8,
            "paddingTop": 4.5,
            "paddingVertical": 5,
          }
        }
      >
        Email
      </Text>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#dddddd",
            "bottom": -0.5,
            "height": 0.5,
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
            "transform": Array [
              Object {
                "scaleX": 0.0001,
              },
            ],
          }
        }
      >
        <View
          hasError={false}
          style={
            Object {
              "backgroundColor": "#6bac43",
              "bottom": 0,
              "height": 0.5,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`The Text Input component should render correctly 1`] = `
<View
  disabled={false}
  style={
    Object {
      "marginVertical": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "flex": 1,
            "opacity": 0,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="Some placeholder"
          placeholderTextColor="rgba(48, 48, 48, 0.6)"
          returnKeyType="done"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 15.75,
              "height": 40,
              "lineHeight": 30,
              "paddingVertical": 5,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
    </View>
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#505050",
            "fontFamily": "Colfax-Bold",
            "fontSize": 15.75,
            "lineHeight": 17.75,
            "paddingBottom": 1.8,
            "paddingTop": 4.5,
            "paddingVertical": 5,
          }
        }
      >
        Some label text
      </Text>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#dddddd",
            "bottom": -0.5,
            "height": 0.5,
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
            "transform": Array [
              Object {
                "scaleX": 0.0001,
              },
            ],
          }
        }
      >
        <View
          hasError={false}
          style={
            Object {
              "backgroundColor": "#6bac43",
              "bottom": 0,
              "height": 0.5,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
